src/Symfony/Component/HttpKernel/Tests/DependencyInjection/RegisterControllerArgumentLocatorsPassTest.php

```diff
@@ -434,6 +435,36 @@ public function testBindWithTarget()
         $expected = ['apiKey' => new ServiceClosureArgument(new Reference('the_api_key'))];
         $this->assertEquals($expected, $locator->getArgument(0));
     }
+
+    public function testAutowireAttribute()
+    {
+        if (!class_exists(Autowire::class)) {
+            $this->markTestSkipped('#[Autowire] attribute not available.');
+        }
+
+        $container = new ContainerBuilder();
+        $resolver = $container->register('argument_resolver.service', 'stdClass')->addArgument([]);
+
+        $container->register('some.id', \stdClass::class);
+        $container->setParameter('some.parameter', 'foo');
+
+        $container->register('foo', WithAutowireAttribute::class)
+            ->addTag('controller.service_arguments');
+
+        (new RegisterControllerArgumentLocatorsPass())->process($container);
+
+        $locatorId = (string) $resolver->getArgument(0);
+        $container->getDefinition($locatorId)->setPublic(true);
+
+        $container->compile();
+
+        $locator = $container->get($locatorId);
+
+        $this->assertInstanceOf(\stdClass::class, $locator->get('foo::fooAction')->get('service1'));
+        $this->assertSame('foo/bar', $locator->get('foo::fooAction')->get('value'));
+        $this->assertSame('foo', $locator->get('foo::fooAction')->get('expression'));
+        $this->assertNull($locator->get('foo::fooAction')->get('service2'));
```

Comment:
This test now fails (it was passing previously). Should I change to:

```suggestion
        $this->assertFalse($locator->get('foo::fooAction')->has('service2'));
```

Reply:
only if it makes sense! I made some suggestions because the current logic is flawed, maybe that's going fix this?

Reply:
I've made the logic changes but this assertion still fails. I don't know enough about how the method is invoked by this system. Should this actually return null or does the system skip if the container doesn't have the service and the arg is nullable?

Reply:
Discovered it's expected for nullable arguments to not be in the service locator.